components/nimbus/src/stateful/nimbus_client.rs

```diff
@@ -684,6 +690,14 @@ impl NimbusClient {
         }
         Ok(())
     }
+
+    fn record_enrollment_status_telemetry(&self) -> Result<()> {
+        for enrollment in self.database_cache.get_enrollments()? {
+            self.host_metrics_handler
+                .record_enrollment_status(enrollment.into());
+        }
+        Ok(())
+    }
```

Comment:
This may be a case of premature optimization: can we avoid going over the FFI more than once by passing the list of enrollments rather than passing them one by one.

Reply:
Eh while you're not wrong in that it _may_ be premature I think it's worth going ahead and updating that.